### PR TITLE
Fix Cannot read property 'text' of undefined crash

### LINKED ISSUE #32358: tsc crashes with TypeError: Cannot read property 'text' of undefined
**TypeScript Version:** 3.6.0-dev.20190711 (also checked on stable 3.5.2)

**Search Terms:**
* TypeError: Cannot read property 'text' of undefined at emitPropertyAccessExpression

**Code**

**test.ts:**
```ts
const fn = (param: string) => undefined;

const foo = {bar: 'a'};
fn(({[foo.bar]: c}) => undefined);
```

```
$ tsc test.ts
```

**Expected behavior:**
`tsc` exits with error `TS2345: Argument of type '({[foo.bar]: c}: any) => void' is not assignable to parameter of type 'string'.` on line 4.

**Actual behavior:**
`tsc` crashes with error:
```
node_modules/typescript/lib/tsc.js:74862
                throw e;
                ^

TypeError: Cannot read property 'text' of undefined
    at emitPropertyAccessExpression (node_modules/typescript/lib/tsc.js:71027:77)
    at pipelineEmitWithHint (node_modules/typescript/lib/tsc.js:70478:32)
    at emitExpression (node_modules/typescript/lib/tsc.js:70140:13)
    at emitComputedPropertyName (node_modules/typescript/lib/tsc.js:70684:13)
    at pipelineEmitWithHint (node_modules/typescript/lib/tsc.js:70207:32)
    at emit (node_modules/typescript/lib/tsc.js:70127:13)
    at emitBindingElement (node_modules/typescript/lib/tsc.js:70999:17)
    at pipelineEmitWithHint (node_modules/typescript/lib/tsc.js:70297:32)
    at emit (node_modules/typescript/lib/tsc.js:70127:13)
    at emitNodeList (node_modules/typescript/lib/tsc.js:72467:21)
```

**Playground Link:** 

http://www.typescriptlang.org/play/#code/MYewdgzgLgBAZmGBeGAKADgQwE6YLYBcM02AlmAOYCUyAfDAK5gAmApnOa8wNwBQvoSLDggQyGAG8ARjiIByTHIC+fBKlQSA2iJAA6GdgC6RYEppJ6TNhzBcqfIA

**Related Issues:** 
Not found.

Original issue: https://github.com/palantir/tslint/issues/4791

### PR DESCRIPTION
**Pull request by rchaser53**
This is a first time for me to send this repository.
I tried to fix the crash bug.
Fixes #32358

### CODE REVIEW

**Comment by rchaser53** [comment]
Sorry, I fixed my code.

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -3527,15 +3527,20 @@ namespace ts {
...
+            let leftStr = hasNoContextSymbolvalueDeclaration(left) ? typeToString(left, left.symbol.valueDeclaration) : typeToString(left);
+            let rightStr = hasNoContextSymbolvalueDeclaration(right) ? typeToString(right, right.symbol.valueDeclaration) : typeToString(right);
             if (leftStr === rightStr) {
                 leftStr = typeToString(left, /*enclosingDeclaration*/ undefined, TypeFormatFlags.UseFullyQualifiedType);
                 rightStr = typeToString(right, /*enclosingDeclaration*/ undefined, TypeFormatFlags.UseFullyQualifiedType);
             }
             return [leftStr, rightStr];
         }
 
+        function hasNoContextSymbolvalueDeclaration(type: Type): boolean {
```
Comment (RyanCavanaugh) [inline src/compiler/checker.ts]: This only operates on `type.symbol`, so can just take the `symbol` directly. Suggested name `symbolValueDeclarationIsContextSensitive` and invert the return value. Implementation then is just
```
   return symbol && symbol.valueDeclaration && isExpression(symbol.valueDeclaration) && isContextSensitive(symbol.valueDeclaration);
```

Comment (rchaser53) [reply src/compiler/checker.ts]: Thank you for the review. Since the behavior changes when the return value is inverted, I changed your suggestion a bit.

**Comment by RyanCavanaugh** [comment]
Thanks!

**Pull request merged**